server/belga/io/feed_parsers/belga_anp_newsml_1_2.py

```diff
@@ -32,22 +32,21 @@ def parser_newsmanagement(self, item, manage_el):
         item['firstcreated'] = item['firstcreated'].astimezone(pytz.utc)
         item['versioncreated'] = item['versioncreated'].astimezone(pytz.utc)
 
-    def parse(self, xml, provider=None):
-        items = super().parse(xml, provider)
-        for item in items:
-            news_products = []
-            for subject in item.get('subject', []):
-                if subject.get('scheme', '') == 'genre':
-                    qcode = subject.get('name')
-                    product = {
-                        'name': self.MAPPING_PRODUCTS.get(qcode, 'GENERAL'),
-                        'qcode': self.MAPPING_PRODUCTS.get(qcode, 'GENERAL'),
-                        'scheme': 'news_products',
-                    }
-                    if product not in item.get('subject', []):
-                        news_products.append(product)
-            item.setdefault('subject', []).extend(news_products)
-        return items
+    def parser_newsitem(self, item, newsitem_el):
+        super().parser_newsitem(item, newsitem_el)
+        product = {}
+        for subject in item.get('subject', []):
+            if subject.get('scheme', '') == 'genre':
+                qcode = self.MAPPING_PRODUCTS.get(subject.get('name'))
+                product = {'name': qcode, 'qcode': qcode, 'scheme': 'news_products'} if qcode else None
+                break
+        if not product:
+            product = {'name': 'GENERAL', 'qcode': 'GENERAL', 'scheme': 'news_products'}
+        item.setdefault('subject', []).append(product)
+        # service is always equal NEWS
```

Comment:
Does it make sense to put it into `BaseBelgaNewsMLOneFeedParser`? 
I see that we perform this action in afp, anp, anpa, efe, iptc7901, tass and so on..